### ISSUE TITLE
Upgrade libsndfile to 1.0.29

### ISSUE THREAD

**Issue by Spekular** [LMMS/lmms#5633]
Issue #5128 was created by a bug in libsndfile, which is fixed in 1.0.29 (via [this commit](https://github.com/erikd/libsndfile/commit/b9e544328e423ac01cac14436b41942e27b6b5aa), I believe). This means #5501 can be reverted if we upgrade to libsndfile 1.0.29.

**Comment by zonkmachine** [comment]
> This means #5501 can be reverted if we upgrade to libsndfile 1.0.29.

Yes, but it will take years for libsndfile 1.0.29 to be adopted. Ubuntu made their latest long-term support release in april this year and it will be maintained until 2025. We need to provide libsndfile ourselves in order to do this. It may not be worth the effort unless there are other issues we see that are fixed by this version as well.

**Comment by Spekular** [comment]
But surely that doesn't mean apt on LTS will be frozen with 2020 versions of everything until 2025? Or am I misunderstanding?

**Comment by zonkmachine** [comment]
> But surely that doesn't mean apt on LTS will be frozen with 2020 versions of everything until 2025?

That is precisely what it means. Ubuntu are notoriously hard to convince to update to a newer version on any release, not only LTS. If they do, it will be it will be in a special updates library that the user will have to change to manually and many are unwilling to do this. Remember how lmms-1.0.0 was rushed into a new Ubuntu stable but the package was accidentally borked? That haunted us for years. I see trouble ahead...

**Comment by eagles051387** [comment]
Afternoon,

For ubuntu that can easily be fixed with a PPA

Regards,
Jonathan

**Comment by zonkmachine** [comment]
> For ubuntu that can easily be fixed with a PPA

And you think we should do that for an issue that already has a workaround?
I suggest we do nothing and bump the topic some four years from now.

**Comment by zonkmachine** [comment]
We could perhaps test libsndfile version and do:

 `clipvalue = -1.0f`

If it's >=1.0.29 ?

http://www.mega-nerd.com/libsndfile/command.html#SFC_GET_LIB_VERSION

**Comment by Rossmaxx** [comment]
update?

**Comment by zonkmachine** [comment]
> update?

Debian, Ubuntu, Fedora, whatever distros are most used. Just look at their latest stable releases and see what version of libsndfile they provide.

**Comment by Rossmaxx** [comment]
Looked it up on repology, unfortunately, 1.0.32 is available on ubuntu 22.04 but everything before that is stuck with 1.0.28

**Comment by zonkmachine** [comment]
Great! Then it will be solved around 26.04 . Let's add a year of margin, so we solve this around 27.04. Bump me then and not before. It's not really an issue if you consider that it's a bit that is gone from the absolute loudest signals and most signals don't clip.
